watertap/unit_models/MD/membrane_distillation_base.py

```diff
@@ -0,0 +1,669 @@
+        self.add_outlet_port(name="cold_ch_outlet", block=self.cold_ch)
+
+        self._add_heat_flux()
+        self._add_mass_flux()
+
+        self.recovery_mass_phase_comp = Var(
+            self.flowsheet().config.time,
+            self.config.hot_ch.property_package.phase_list,
+            self.config.hot_ch.property_package.component_list,
+            initialize=0.1,
+            bounds=(1e-11, 0.99),
+            units=pyunits.dimensionless,
+            doc="Mass-based component recovery",
+        )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.config.hot_ch.property_package.component_list,
+        )
+        def eq_recovery_mass_phase_comp(b, t, j):
+            return (
+                b.recovery_mass_phase_comp[t, "Liq", j]
+                * b.hot_ch.properties[t, b.first_element].flow_mass_phase_comp["Liq", j]
+                == b.cold_ch.properties[t, b.first_element].flow_mass_phase_comp[
+                    "Liq", j
+                ]
+                - b.cold_ch.properties[
+                    t, b.cold_ch.length_domain.last()
+                ].flow_mass_phase_comp["Liq", j]
+            )
+
+        self._add_mass_transfer()
+        self._add_heat_transfer()
+
+    def _add_mass_flux(self):
+
+        solvent_set = self.config.hot_ch.property_package.solvent_set
+        solute_set = self.config.hot_ch.property_package.solute_set
+
+        units_meta = (
+            self.config.hot_ch.property_package.get_metadata().get_derived_units
+        )
+
+        # todo: add calculation method for permeability coefficient
+        self.permeability_coef = Var(
+            self.flowsheet().config.time,
+            initialize=1e-10,
+            bounds=(1e-11, 1e-9),
+            units=units_meta("mass")
+            * units_meta("time") ** -1
+            * units_meta("length") ** -1
+            * units_meta("pressure") ** -1,
+            doc="membrane permeabiity coeffcient",
+        )
+
+        self.flux_mass = Var(
+            self.flowsheet().config.time,
+            self.difference_elements,
+            initialize=1e-3,
+            bounds=(1e-6, 10),
+            units=units_meta("mass")
+            * units_meta("time") ** -1
+            * units_meta("length") ** -2,
+            doc="solvent vapor mass flux across the membrane",
+        )
+
+        self.flux_enth_hot = Var(
+            self.flowsheet().config.time,
+            self.difference_elements,
+            initialize=1e3,
+            bounds=(1e-4, 1e5),
+            units=pyunits.J * pyunits.s**-1 * pyunits.m**-2,
+            doc="hot side evaporation enthalpy flux",
+        )
+
+        self.flux_enth_cold = Var(
+            self.flowsheet().config.time,
+            self.difference_elements,
+            initialize=1e3,
+            bounds=(1e-4, 1e5),
+            units=pyunits.J * pyunits.s**-1 * pyunits.m**-2,
+            doc="cold side condensation enthalpy flux",
+        )
```

Comment:
Can't these go in one of the channel files? Then, you could later reference `hot_ch.flux_enth` or `cold_ch.flux_enth`

Reply:
Only these two variables, but not their associated constraints, may go to MD_channel_base.